### PR TITLE
[FIX] mrp: compare float thanks to `float_compare`

### PR DESCRIPTION
**Pull request by adwid**
In some situations, the user can't redefine the quantity to produce of a
MO

To reproduce the issue:
(Use demo data. Enable debug mode)
1. In Decimal Accuracy, edit Product Unit of Measure:
    - Digits: 5
2. Create a MO:
    - Product: [FURN_7023] Wood Panel
    - Quantity To Produce: 1000
3. Produce 800
4. Update the Quantity To Produce: 800

Error: An error message is displayed: "You have already processed
800.00000. Please input a quantity higher than 800.00000"

Due to a floating point issue, the produced quantity is actually
800.0000000000001 which is greater than 800. This is the reason why the
`UserError` is raised

Some similar issues might be found with the other float comparisons in
`change_prod_qty`.

OPW-2689831

### CODE REVIEW

**Comment by adwid** [comment]
Not reproducible in 14.0
@fw-bot ignore

File: addons/mrp/wizard/change_production_qty.py [hunk 1]
```diff
@@ -98,14 +98,14 @@ def change_prod_qty(self):
                 if production.product_id.tracking == 'serial':
                     quantity = 1.0 if not float_is_zero(quantity, precision_digits=precision) else 0.0
                 else:
-                    quantity = quantity if (quantity > 0) else 0
+                    quantity = quantity if float_compare(quantity, 0, precision_digits=precision) > 0 else 0
```
Comment (amoyaux) [inline addons/mrp/wizard/change_production_qty.py:101]: We should use the mo_id.product_uom_id.rounding in that case no?

File: addons/mrp/wizard/change_production_qty.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ def change_prod_qty(self):
         for wizard in self:
             production = wizard.mo_id
             produced = sum(production.move_finished_ids.filtered(lambda m: m.product_id == production.product_id).mapped('quantity_done'))
-            if wizard.product_qty < produced:
+            if float_compare(wizard.product_qty, produced, precision_digits=precision) < 0:
```
[the remark this answers is not in the document]
Comment (adwid) [reply addons/mrp/wizard/change_production_qty.py:49]: I didn't check for `wizard.product_qty`, but `produced` is based on `quantity_done` which uses the general precision:
https://github.com/odoo/odoo/blob/317f02071d8d285e0b59fbb6a662289008f22606/addons/stock/models/stock_move.py#L164

File: addons/mrp/wizard/change_production_qty.py (hunk 1 above)
Comment (adwid) [reply addons/mrp/wizard/change_production_qty.py:101]: Same here, I'm not sure. Considering the definition of `wo.qty_produced` (which is used to compute `quantity`):
https://github.com/odoo/odoo/blob/81497125d8100c6bdd2dc30434232a88a419a3e3/addons/mrp/models/mrp_workorder.py#L49-L53

**Comment by amoyaux** [comment]
robodoo r+

**Pull request closed without merging**